Formula/d/docker.rb

```diff
@@ -35,7 +34,7 @@ def install
     ldflags = %W[
       -s -w
       -X github.com/docker/cli/cli/version.BuildTime=#{time.iso8601}
-      -X github.com/docker/cli/cli/version.GitCommit=#{Utils.git_short_head}
+      -X github.com/docker/cli/cli/version.GitCommit=#{tap.user}
```

Comment:
We can probably keep this in case users rely on it. But we can definitely install the completions from a tarball.

Reply:
why does people reply on it? what is the use case?

Reply:
Debugging, perhaps? Presumably if there were no use case upstream wouldn't build this into the output of their version information...

Reply:
for release build, it is not useful (the [vcs provider introduced in go1.18](https://tip.golang.org/doc/go1.18#debug/buildinfo) for debugging build), that is why I am removing it for all golang builds.

Reply:
I've seen quite some arguments from upstream developers that disagree with that. They generally want to know what commit the versioned build was based on.

Reply:
We should definitely keep this. I see no reason to ever override `GitCommit` to not mention a Git commit and, if this has been done on any other formulae, these changes should be reverted.

```suggestion
      -X github.com/docker/cli/cli/version.GitCommit=#{Utils.git_short_head}-Homebrew
```

or something if you really want to identify Homebrew usage here.

Reply:
> I've seen quite some arguments from upstream developers that disagree with that. They generally want to know what commit the versioned build was based on.

The version is equivalent to the commit, if we are going to do it, should we revert all the go formulae to the git checkout. (maybe another RFC on this topic?)

Reply:
I don't think it is equivalent.



> should we revert all the go formulae to the git checkout.

or query it from the API?

Reply:
for now, I just revert the docker build change